src/capi/typeobject.cpp

```diff
@@ -2993,6 +2993,17 @@ void commonClassSetup(BoxedClass* cls) {
     assert(cls->tp_dict && cls->tp_dict->cls == attrwrapper_cls);
 }
 
+static void checkIfExceptionType(BoxedClass* cls) {
```

Comment:
is there a constant time check we can perform here using `Py_TPFLAGS_BASE_EXC_SUBCLASS` ?  I'm not sure if we do the right setup 100% of the time for it, but if we do, it should be quite a bit quicker.